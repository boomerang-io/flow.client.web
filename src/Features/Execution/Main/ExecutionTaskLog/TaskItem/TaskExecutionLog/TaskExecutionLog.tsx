import React from "react";
import cx from "classnames";
import { Button, ComposedModal } from "@boomerang-io/carbon-addons-boomerang-react";
import { Toggle } from "@boomerang-io/carbon-addons-boomerang-react";
import { ModalBody } from "@boomerang-io/carbon-addons-boomerang-react";
import { LazyLog, ScrollFollow } from "react-lazylog";
import { serviceUrl } from "Config/servicesConfig";
import { PRODUCT_SERVICE_ENV_URL } from "Config/servicesConfig";
import styles from "./taskExecutionLog.module.scss";

const DEV_STREAM_URL =
  "https://gist.githubusercontent.com/helfi92/96d4444aa0ed46c5f9060a789d316100/raw/ba0d30a9877ea5cc23c7afcd44505dbc2bab1538/typical-live_backing.log";

type Props = {
  flowActivityId: string;
  flowTaskId: string;
  flowTaskName: string;
};

export default function TaskExecutionLog({ flowActivityId, flowTaskId, flowTaskName }: Props) {
  const [follow, setFollow] = React.useState(true);
  const [error, setError] = React.useState(false);

  return (
    <ComposedModal
      composedModalProps={{
        containerClassName: styles.container,
        shouldCloseOnOverlayClick: true,
      }}
      modalHeaderProps={{
        title: "Execution Log",
        label: `${flowTaskName}`,
      }}
      modalTrigger={({ openModal }: { openModal: () => void }) => (
        <Button className={styles.trigger} kind="ghost" size="small" onClick={openModal}>
          View Log
        </Button>
      )}
    >
      {() => (
        <ModalBody>
          <ScrollFollow
            startFollowing={true}
            render={({ onScroll }: { onScroll: () => void }) => (
              <>
                <div className={styles.followToggle}>
                  <label
                    className={cx(styles.followToggleText, { [styles.disabled]: Boolean(error) })}
                    htmlFor="task-log-toggle"
                  >
                    Follow log
                  </label>
                  <Toggle
                    defaultValue={follow}
                    disabled={!!error}
                    id="task-log-toggle"
                    onChange={() => setFollow(!follow)}
                    toggled={follow}
                  />
                </div>
                <LazyLog
                  enableSearch={true}
                  fetchOptions={
                    PRODUCT_SERVICE_ENV_URL.includes("localhost") ? { credentials: "omit" } : { credentials: "include" }
                  }
                  follow={follow}
                  onScroll={onScroll}
                  onError={(err: boolean) => setError(err)}
                  selectableLines={true}
                  stream={true}
                  url={
                    PRODUCT_SERVICE_ENV_URL.includes("localhost")
                      ? DEV_STREAM_URL
                      : serviceUrl.getWorkflowExecutionLog({ flowActivityId, flowTaskId })
                  }
                />
              </>
            )}
          />
        </ModalBody>
      )}
    </ComposedModal>
  );
}
