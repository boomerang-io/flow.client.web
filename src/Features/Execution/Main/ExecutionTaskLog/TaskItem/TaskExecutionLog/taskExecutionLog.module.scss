@import "Styles/mixins";

.trigger {
  margin-left: -0.75rem;
}

.container {
  max-width: 70rem;
  max-height: 50rem;

  :global .cds--modal-content {
    padding-right: 1rem;
  }

  h1 {
    margin: 0;
  }

  :global .cds--modal-content {
    height: 100%;
  }

  .bmrg--b-modal-flow-header {
    margin: 1.875rem auto;
  }

  .bmrg--c-modal-flow-content {
    margin-bottom: 2rem;
  }

  :global .react-lazylog-searchbar {
    background: var(--cds-text-01);
    padding: 0.625rem;
    font-family: var(--cds-font-family);
  }

  :global .react-lazylog-searchbar-input {
    border-radius: 0.125rem;
    color: var(--cds-ui-01);
    letter-spacing: 0.0625rem;
    font-family: var(--cds-font-family);
    font-size: 0.75rem;
    &:focus {
      @include focus-outline();
    }
    &:active {
      outline: none;
    }
  }

  :global .react-lazylog-searchbar-filter {
    &:focus {
      @include focus-outline();
    }
  }

  :global .react-lazylog-searchbar-matches {
    color: var(--cds-ui-01);
    font-family: var(--cds-font-family);
    font-size: 0.75rem;
    letter-spacing: 0.0625rem;
    min-width: 5rem;
  }

  :global .react-lazylog {
    background: var(--cds-text-01);
    color: var(--cds-ui-01);
    padding: 1rem 1rem 1rem 0;
    font-family: "IBM Plex Mono", monospace;
    font-size: 0.75rem;
    line-height: 1.5;
  }

  .followToggle {
    position: absolute;
    right: 358px;
    top: 1rem;
  }

  .followToggleText {
    color: var(--cds-ui-01);
    font-size: 0.75rem;
    font-weight: 400;
    letter-spacing: 0.0625rem;
    margin-right: 0.5rem;
    white-space: nowrap;
    margin-bottom: 0.375rem;
    &.disabled {
      opacity: 0.5;
    }
  }
}
