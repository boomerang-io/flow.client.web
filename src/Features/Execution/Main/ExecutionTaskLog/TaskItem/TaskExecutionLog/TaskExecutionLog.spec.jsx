import React from "react";
import TaskExecutionLog from "./index";

const props = {
  flowActivityId: "1",
  flowTaskId: "2",
  flowTaskName: "Send Slack Message"
};

describe("TaskExecutionLog --- Snapshot", () => {
  it("Capturing Snapshot of TaskExecutionLog", () => {
    const { baseElement } = rtlRender(<TaskExecutionLog {...props} />);
    expect(baseElement).toMatchSnapshot();
  });
});
