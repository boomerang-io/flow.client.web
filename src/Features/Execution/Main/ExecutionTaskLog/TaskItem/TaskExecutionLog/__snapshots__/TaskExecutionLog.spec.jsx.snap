// Vitest Snapshot v1

exports[`TaskExecutionLog --- Snapshot > Capturing Snapshot of TaskExecutionLog 1`] = `
<body
  id="app"
>
  <div>
    <button
      class="bx--btn bx--btn--sm bx--btn--ghost"
      tabindex="0"
      type="button"
    >
      View Log
    </button>
  </div>
  <div
    class="bx--bmrg-modal-portal"
  />
</body>
`;
